tests/dialects/test_csl_stencil.py

```diff
@@ -0,0 +1,56 @@
+from xdsl.builder import Builder
+from xdsl.dialects.builtin import IntegerAttr, IntegerType, MemRefType, TensorType, f32
+from xdsl.dialects.csl.csl_stencil import AccessOp, ApplyOp
+from xdsl.dialects.stencil import IndexAttr, TempType
+from xdsl.ir import Region, SSAValue
+from xdsl.utils.test_value import TestSSAValue
+
+
+def test_access_patterns():
+    temp_t = TempType(5, f32)
+    temp = TestSSAValue(temp_t)
+    mref = TestSSAValue(mref_t := MemRefType(tens_t := TensorType(f32, (5,)), (4,)))
+
+    @Builder.implicit_region((mref_t, temp_t))
+    def region0(args: tuple[SSAValue, ...]):
+        t0, t1 = args
+        for x in (-1, 1):
+            AccessOp(t0, IndexAttr.get(x, 0), tens_t)
+        for y in (-1, 1):
+            AccessOp(t0, IndexAttr.get(0, y), tens_t)
+
+        AccessOp(t1, IndexAttr.get(1, 1), tens_t)
+        AccessOp(t1, IndexAttr.get(-1, -1), tens_t)
+
+    @Builder.implicit_region((temp_t, temp_t))
+    def region1(args: tuple[SSAValue, ...]):
+        t0, t1 = args
+        AccessOp(t0, IndexAttr.get(0, 0), tens_t)
+        AccessOp(t1, IndexAttr.get(0, 0), tens_t)
```

Comment:
What happens with the accesses if this doesn't just access (0,0)?

Reply:
Since the `get_accesses` function loops over
```
for arg in self.chunk_reduce.block.args + self.post_process.block.args:
```
there should in theory be no difference between whether a particular access is in the first or second region, so this is merely testing that this works, without re-doing the more in-depth testing for the first region.